Paint/Paint/MainForm.cs

```diff
@@ -0,0 +1,153 @@
+        {
+            if (SaveNewImageDialog.ShowDialog() == DialogResult.Cancel) return;
+            CanvasPB.Image?.Save(SaveNewImageDialog.FileName);   
+        }
+        
+        private void CoordinatesSetter(object sender, MouseEventArgs eventArg)
+        {
+            CoordinatesLabel.Text = $@"({eventArg.X}, {eventArg.Y})";
+        }
+        
+        private void CoordinatesClearer(object sender, EventArgs eventArg)
+        {
+            CoordinatesLabel.Text = "";
+        }
+
+        private void SetNewColor(object sender, EventArgs eventArg)
+        {
+            ColorPickerDialog.ShowDialog();
+            currentPenColor = ColorPickerDialog.Color;
+            ColorPickerBtn.BackColor = currentPenColor;
+            drawingPen.Color = currentPenColor;
+        }
+
+        private void DrawByPressedPen(object sender, MouseEventArgs eventArg)
+        {
+            if (isPenDown && lastPoint != Point.Empty)
+            {
+                using (Graphics g = Graphics.FromImage(CanvasPB.Image))
+                {
+                    g.DrawLine(drawingPen, lastPoint, eventArg.Location);
+                }
+                CanvasPB.Invalidate();
+                lastPoint = eventArg.Location;
+            }
+        }
+
+        private void OnMouseDown(object sender, MouseEventArgs eventArg)
+        {
+            isPenDown = true;
+            lastPoint = eventArg.Location;
+        }
+        
+        private void OnMouseUp(object sender, MouseEventArgs eventArg)
+        {
+            isPenDown = false;
+            lastPoint = Point.Empty;
+        }
+
+        private void UpdateImage()
+        {
+            CanvasPB.Image = new Bitmap(lastImage);
+            CanvasPB.Size = new Size(lastImage.Width, lastImage.Height);
+        }
+        
+        private void UpdatePen(object sender, EventArgs e)
+        {
+            drawingPen.Color = currentPenColor;
+            penWidth = (int)PenSizeUD.Value;
+            drawingPen.Width = penWidth;
+        }
+        
+        private Point ShowCustomDialog()
+        {
+            Form prompt = new Form();
+            prompt.Width = 200;
+            prompt.Height = 150;
+            prompt.Text = "New Image";            
+            prompt.AutoSize = false;
+            prompt.MaximizeBox = false;
+            prompt.FormBorderStyle = FormBorderStyle.FixedSingle;
+            var widthLabel = new Label() { Left = 10, Top=20, Text="Width", Width = 60};
+            var widthUD = new NumericUpDown () { Left = 100, Top=20, Width=70, Minimum = 0, Maximum = Decimal.MaxValue};
+            
+            var heightLabel = new Label() { Left = 10, Top=40, Text="Height", Width = 60 };
+            var heightUD = new NumericUpDown () { Left = 100, Top=40, Width=70, Minimum = 0, Maximum = Decimal.MaxValue};
+            
+            var confirmation = new Button() { Text = "Ok", Left=40, Width=100, Top=70 };
+            
```

Comment:
Нужно превратить это окно в настоящее диалоговое окно, чтобы изменения происходили только при нажатии кнопки "ОК".  Сейчас результат нажатия кнопки не учитывается, можно открыть диалог, ввести значения, закрыть крестиком, а значения все равно передадутся. Делаем кнопку "ОК" настоящей кнопкой, по которой пользователь принимает введенные значения:
```suggestion
            prompt.AcceptButton = confirmation;
            confirmation.DialogResult = DialogResult.OK;
```